rinex-cli/src/positioning/cggtts/mod.rs

```diff
@@ -84,6 +84,20 @@ where
     let nav_data = ctx.data.nav_data().unwrap();
     let meteo_data = ctx.data.meteo_data();
 
+    let clk_data = ctx.data.clk_data();
+    let has_clk_data = clk_data.is_some();
+
+    let sp3_data = ctx.data.sp3_data();
+
+    let sp3_has_clock = if has_clk_data {
+        false // always prefer CLK product
+    } else {
+        match sp3_data {
```

Comment:
Not tested, but you can also do `sp3.sv_clock().map(Vec::is_empty).unwrap_or(false)`.

Reply:
I don't understand what `.map(Vec::is_empty)` really does, it's not like a typical `.map(|x|)` lambda.

> but you can also do sp3.sv_clock().map(Vec::is_empty).unwrap_or(false)

Kind of, except that `sp3` is actually `Option<sp3>` here 

I tried something like this

```rust
} else {
     sp3_data
          .unwrap_or(false) // if not present, well.. no need to bother
          .sv_clock()
          .map(Vec::is_empty) // don't understand this
          .unwrap_or(false)
}
```

Reply:
`.map(Vec::is_empty)` is more or less the same thing as `.map(|x: Vec<_>| x.is_empty())`.

Your snippet is not going to work because you're `sp3_data.unwrap_or(false)` can't be both a `bool` and whatever `sp3_data.unwrap()` is.

Reply:
> .map(Vec::is_empty) is more or less the same thing as .map(|x: Vec<_>| x.is_empty()).

👍 

> Your snippet is not going to work because you're sp3_data.unwrap_or(false) can't be both a bool and whatever sp3_data.unwrap() is.


That's indeed correct